app/models/report.rb

```diff
@@ -3,15 +3,50 @@
 class Report < ApplicationRecord
   belongs_to :user
   has_many :comments, as: :commentable, dependent: :destroy
+  has_many :active_relationships, class_name: 'Relationship', foreign_key: 'mentioning_id', inverse_of: :mentioning, dependent: :destroy
+  has_many :passive_relationships, class_name: 'Relationship', foreign_key: 'mentioned_id', inverse_of: :mentioned, dependent: :destroy
+  has_many :mentioning_reports, through: :active_relationships, source: :mentioned
+  has_many :mentioned_reports, through: :passive_relationships, source: :mentioning
 
   validates :title, presence: true
   validates :content, presence: true
 
+  after_create :save_new_mentions
```

Comment:
```suggestion
  after_create :update_mentions
```

としても実質的にやってることは同じになりませんかね？

Reply:
同じになります🙇🏻‍♀️
修正しました。